[Neo Heng Kai] iP

KiKuasaurus opened the pull request:
# Ontos

>  “The philosophers have only interpreted the world, in various ways. The point, however, is to change it." -- Karl Marx\([Source](https://www.goodreads.com/author/quotes/7084.Karl_Marx)\)

## Ontos is...
* Easy to use
* Hassle free

## All you need to do is...
1. Download the file from [here](https://github.com/KiKuasaurus/ip/releases/tag/A-Jar)
2. Run the commands
```
java -jar ontos.jar
```

## Here are some sample commands
```
todo Finish homework
deadline Complete project /by 2023-09-30
event Team meeting /from 2023-09-15 /to 2023-09-15
list
mark 2
unmark 2
delete 1
find homework
bye
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

KiKuasaurus committed:
level-0 : Rename, Greet, Exit

KiKuasaurus committed:
Delete src/main/java/Duke.java

removed original Duke.java for renamed file

KiKuasaurus committed:
Level-1: Echo

KiKuasaurus committed:
Level-2: Add, List

KiKuasaurus committed:
Level-2: Add, List, changed colon to dot when listing

KiKuasaurus committed:
Level-3: Mark as Done

KiKuasaurus committed:
level-3: Mark as Done

KiKuasaurus committed:
Level-4: ToDo, Event, Deadline

KiKuasaurus committed:
Level-5: Handle Errors

KiKuasaurus committed:
Level-5: Handle Errors, fixed formatting for some errors

KiKuasaurus committed:
Level-6: Delete

KiKuasaurus committed:
Merge branch 'master' of https://github.com/KiKuasaurus/ip

KiKuasaurus committed:
A-TextUiTesting: Automated Text UI Testing

KiKuasaurus committed:
fixed a bug relating to unmark

KiKuasaurus committed:
fix formatting

KiKuasaurus committed:
Implement Level-7: Save

KiKuasaurus committed:
Merge pull request #1 from KiKuasaurus/branch-Level-7

LGTM

KiKuasaurus committed:
Done A-MoreOOP: Use More OOP

KiKuasaurus committed:
Merge pull request #2 from KiKuasaurus/branch-A-MoreOOP

LGTM

KiKuasaurus committed:
Done Level-8: Dates and Times

KiKuasaurus committed:
Merge pull request #3 from KiKuasaurus/branch-Level-8

LGTM

KiKuasaurus committed:
Do A-Packages: Organize into Packages

KiKuasaurus committed:
Do A-Packages: Organize into Packages

KiKuasaurus committed:
Merge pull request #4 from KiKuasaurus/branch-A-Packages

lgtm

KiKuasaurus committed:
Merge pull request #5 from KiKuasaurus/add-gradle-support

Add gradle support

KiKuasaurus committed:
Add Main and move Ontos.java into the package

KiKuasaurus committed:
A-Gradle: Use Gradle

KiKuasaurus committed:
Merge pull request #6 from KiKuasaurus/branch-A-Gradle

LGTM

KiKuasaurus committed:
Do A-JUnit

KiKuasaurus committed:
Merge pull request #7 from KiKuasaurus/branch-A-JUnit

Do A-JUnit

KiKuasaurus committed:
Do A-JavaDoc: JavaDoc

KiKuasaurus committed:
Do Level-9: Find

KiKuasaurus committed:
Merge pull request #11 from KiKuasaurus/branch-A-JavaDoc

Do A-JavaDoc: JavaDoc

KiKuasaurus committed:
Merge branch 'master' into branch-Level-9

KiKuasaurus committed:
Merge pull request #10 from KiKuasaurus/branch-Level-9

Do Level-9: Find

KiKuasaurus committed:
Added image for comment

KiKuasaurus committed:
Delete chatbot.webp to replace it with a jpeg file

KiKuasaurus committed:
Add new image for comment

KiKuasaurus committed:
Do A-CheckStyle

KiKuasaurus committed:
Merge pull request #12 from KiKuasaurus/branch-A-CheckStyle

LGTM

KiKuasaurus committed:
Do Level-10: GUI

KiKuasaurus committed:
Do Level-10: GUI

KiKuasaurus committed:
Merge pull request #13 from KiKuasaurus/branch-Level-10

LGTM

ruiming97 commented on src/main/java/ontos/Ontos.java:
```diff
@@ -0,0 +1,88 @@
+     * and manages the task list until the user exits.
+     */
+    public void run() {
+        ui.showLine();
```
I like how you use a separate method to print the divider line

ruiming97 commented on src/main/java/ontos/storage/SaveManager.java:
```diff
@@ -0,0 +1,136 @@
+ * Reads and writes the task list from and to storage.
+ */
+public class SaveManager {
+    protected static HashMap<Character, Boolean> isDoneHashMap = new HashMap<>() {{
```
I like how you use a HashMap to store the string representation of the task status. Any particular reason that you're doing it this way instead of simply using an if-else statement that checks if the character is a '0' and '1'?

ruiming97 commented on src/main/java/ontos/task/Task.java:
```diff
@@ -0,0 +1,344 @@
+     * @param isDone      Whether the task is completed.
+     * @return A new ToDo Task.
+     */
+    public static Task toDo(String description, boolean isDone) {
```
I like how you create another factory method to specify completion status of the task.

ruiming97 commented on src/main/java/ontos/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+        for (int i = 0; i < this.getSize(); i++) {
+            int index = i + 1;
+            output.append(" ").append(index).append(". ").append(taskList.get(i).toString());
+            if (index < this.getSize()) {
```
I like how you use an if-else statement to avoid adding an empty line at the end of the file.

ruiming97 commented on src/main/java/ontos/ui/Ui.java:
```diff
@@ -0,0 +1,231 @@
+ */
+public class Ui {
+    /** Greeting message displayed when the application starts. */
+    private static String hello = " Hello! I'm Ontos \n What can I do for you?";
```
Should all the messages be declared as private static final like constants and named in all caps with a common prefix like MESSAGE_HELLO?

Airiinnn replied:
As someone else has mentioned, it might be more appropriate to use constants.

ruiming97 commented on src/main/java/ontos/ui/Ui.java:
```diff
@@ -0,0 +1,231 @@
+     * @return the message as a String.
+     */
+    public String taskDoesNotExist() {
+        String output = " I'm sorry, but this task doesn't exist.";
```
Is there a reason why this message is not defined in the header?

ruiming97 reviewed (commented):
Overall the code is very well-written with detailed java docs.

Airiinnn commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    public static DialogBox getOntosDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
nit: In this small context might be fine, but `db` can be misleading as it is usually interpreted as database.

Airiinnn commented on src/main/java/ontos/Ontos.java:
```diff
@@ -0,0 +1,88 @@
+     */
+    public Ontos(String saveLocation) {
+        Path projectRoot = Paths.get("").toAbsolutePath().getParent().getParent().getParent();
+        this.ui = new Ui();
```
The coding standard recommends avoiding unnecessary use of `this` with fields.

https://se-education.org/guides/tutorials/intellijCodeStyle.html

Airiinnn commented on src/main/java/ontos/commands/Command.java:
I like your use of OOP for handling the different commands. Perhaps it might be better to separate the individual commands into separate files?

Airiinnn reviewed (commented):
Great work so far. A few suggestions for coding standard & quality, but otherwise very good job!

KiKuasaurus committed:
Do A-Assertions

KiKuasaurus committed:
Do A-CodeQuality

KiKuasaurus committed:
Merge pull request #14 from KiKuasaurus/branch-A-Assertions

Do A-Assertions

KiKuasaurus committed:
Revert "Do A-Assertions"

KiKuasaurus committed:
Merge pull request #15 from KiKuasaurus/revert-14-branch-A-Assertions

Revert "Do A-Assertions"

KiKuasaurus committed:
Revert "Revert "Do A-Assertions""

KiKuasaurus committed:
Merge pull request #16 from KiKuasaurus/revert-15-revert-14-branch-A-Assertions

Revert "Revert "Do A-Assertions""

KiKuasaurus committed:
Merge branch 'master' into branch-A-CodeQuality

KiKuasaurus committed:
Fix a test case that was broken due to assertions

KiKuasaurus committed:
Merge pull request #17 from KiKuasaurus/branch-A-CodeQuality

LGTM

KiKuasaurus committed:
Do BCD-Extension

KiKuasaurus committed:
Merge pull request #18 from KiKuasaurus/branch-BCD-Extension

LGTM

KiKuasaurus committed:
Remove static strings in ui

There were several Strings that were cached in Ui, however it was not
consistent with all the strings.

Changing it to have all Strings initialised during method calls makes
it more consistent and improves code quality.

In a bid to improve consistency, let's move all String so that they
are only initialised when the method is called rather than it being
cached on object creation.

KiKuasaurus committed:
Fix nesting issues for loadSave() in SaveManager

The method loadSave() had nesting issues.

Fixing loadSave() improves code quality.

To reduce the nesting issues, a method addTask()
was extracted to make it cleaner.

KiKuasaurus committed:
Merge pull request #19 from KiKuasaurus/branch-A-FullCommitMessage

LGTM

KiKuasaurus committed:
Add help functionality

KiKuasaurus committed:
Merge pull request #21 from KiKuasaurus/branch-help

LGTM

KiKuasaurus committed:
Change a local variable name to be less ambiguous

KiKuasaurus committed:
Fix formatting for save location request

KiKuasaurus committed:
Update README for new name and initial prompt\

KiKuasaurus committed:
Add Ui.png

KiKuasaurus committed:
Fix a bug related to the help function

KiKuasaurus committed:
Do A-Userguide

KiKuasaurus committed:
Fix an issue where the user guide was not complete

KiKuasaurus committed:
Fix an issue where ui image was not showing up
